sdk/src/loom/Platform/AppData.ls

```diff
@@ -0,0 +1,346 @@
+            {
+                init();
+            }
+
+            load();
+		}
+
+
+        /**
+        Called when the before the JSON file is created for the first time
+         * Designed to be overwritten by the child object to handle custom actions.
+         */
+        protected function init():void {}
+
+
+        /**
+        Saves the local JSON data object to the system as the AppData.
+         * Designed to be optionally overwritten by the child object to handle custom actions.
+         *
+         *  @return Boolean Was the JSON data file saved successfully or not
+         */
+        protected function save():Boolean
+        {
+            //save the JSON
+            if(!File.writeTextFile(_writePath, jsonData.serialize()))
+            {
+                Logger.warn(this, "save", "Error saving AppData JSON: " + _writePath);
+                return false;
+            }
+            trace("App Data saved to: " + _writePath);            
+            return true;
+        }
+
+
+        /**
+        Loads the JSON file for the AppData from the system into a locally stored JSON data object.
+         * Designed to be optionally overwritten by the child object to handle custom actions.
+         *
+         *  @return Boolean Was the JSON data file loaded successfully or not
+         */
+		protected function load():Boolean
+		{
+            //load the JSON into a text file
+            if(!File.fileExists(_writePath))
+            {
+                Logger.print(this, "No AppData JSON found at: " + _writePath);
+                return false;
+            }
+            var textFile:String = File.loadTextFile(_writePath);
+            if((textFile == null) || (textFile == ""))
+            {
+                Logger.warn(this, "load", "Invalid contents of AppData JSON file: " + _writePath);
+                return false;
+            }
+            trace("App Data loaded from: " + _writePath);            
+
+            //create JSON from text file data
+            jsonData = new JSON();
+            return jsonData.loadString(textFile);
+		}
+
+
+        /**
+        Clears the locally read in JSON data, but leave the file on the system as-is.  
+         * Designed to be optionally overwritten by the child object to handle custom actions.
+         */
+		protected function clear():void
+		{
+            //clear the cached JSON data
+            jsonData = new JSON();
+            jsonData.loadString("{}");
+		}
+
+
+        /**
+        Removes the JSON file storing the AppData from the system and clears the local data as well.
+         */
+        public function purge():void
+        {
+            //check for valid file
+            if(!String.isNullOrEmpty(_writePath))
+            {
+                if(File.fileExists(_writePath))
+                {
+                    //delete the file
+                    File.removeFile(_writePath);
+                    trace("App Data file removed: " + _writePath);            
```

Comment:
Yeah, I specifically made the trace() calls in this file be for logging purposes so that people can see if their stuff has written properly, etc.  There are other Logger calls for more error/debug stuff, but I thought it would be good to keep output specific to file save/load/delete permanently there.